### PR TITLE
swift-api-checker: teach infer-imports to find all catalyst-supporting Swift frameworks

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: utils/api_checker/sdk-module-lists/infer-imports.py [hunk 1]
```diff
@@ -15,6 +15,31 @@ def get_immediate_subdirectories(a_dir):
             if os.path.isdir(os.path.join(a_dir, name))]
 
 
+def collect_catalyst_frameworks(frameworks_path):
+    names = []
+    for frame in os.listdir(frameworks_path):
+        if frame.endswith(".framework"):
+            name = frame[:-len(".framework")]
+            module_dir_path = frameworks_path + '/' + frame + '/Modules'
```
Comment (akyrtzi) [inline utils/api_checker/sdk-module-lists/infer-imports.py]: The python idiomatic way to concatenate path components is using `os.path.join`, like this: `os.path.join(frameworks_path, frame, Modules)`. You can pass any number of arguments.

Comment (nkcsgexi) [reply utils/api_checker/sdk-module-lists/infer-imports.py]: Fixed.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by akyrtzi — APPROVED** [review]

**Pull request merged**